include/finufft_opts.h

```diff
@@ -5,6 +5,30 @@
 #ifndef FINUFFT_OPTS_H
 #define FINUFFT_OPTS_H
 
+// Marco Barbone: 5.8.2024
+// These are user-facing to that the user can reset to the default value
```

Comment:
The user would have to recompile finufft lib to change these, correct? Couldn't they also just change finufft_default_opts() function themselves instead?

Reply:
To change the MACROS? They are not supposed to be changed.
I think having values lying around instead of a named macro or constant is error prone. If we start adding more or removing options this can get out of have. I would like to pay the price now and have something like this now that delay to the future and let the dust pile. @blackwer what do you think?

Reply:
I'm on the fence. The macros are a lot of pollution, but it's nice having everything in one place -- especially when you need it multiple times. My inclination is to make a `constexpr finufft_opts FINUFFT_DEFAULT_OPTS{...};` in a header so that it can be autodocced and essentially namespace itself, grouping all relevant opts under the same header (alternatively prefixing `FINUFFT_DEFAULT_OPTS_{OPTION}` is reasonable, if we fix the doc issue). with the `constexpr`, `finufft_default_opts()` can just copy it. when you need a comparison it won't pay a cost of lookup (as compared to a static variable somewhere). there might be a problem I'm overlooking, but thoughts?